### PR TITLE
[IOP] : Adding New Metrics controllers to push metrics to prometheus

### PR DESCRIPTION
**Pull request by vinaybheri**
* Currently the sfserviceinstance metrics are sent from sfserviceinstancereplicator controller. For better channelization and readability now we have created new controller sfserviceinstancemetrics to push  sfserviceinstance metrics to prometheus.
* Similarly till now the sfservicebinding metrics are sent from sfservicebindingreplicator controller. For better channelization and readability now we have created a new controller sfservicebindingmetrics to push  sfservicebinding metrics to prometheus.

### CODE REVIEW

**Review by jintusebastian — CHANGES_REQUESTED** [review]
Looks like some commits are unrelated to this PR: 762c3598726a85d1aee91f529658460f504de421

Please rebase the branch with master

**Comment by vinaybheri** [comment]
@jintusebastian Updated changes with latest master branch changes.
Kindly check and review.

File: interoperator/controllers/multiclusterdeploy/sfservicebindingmetrics/sfservicebindingmetrics_controller.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			"state",
+			"sf_namespace",
+			//"last_operation",
+		},
+	)
+)
+
+var getWatchChannel = watchmanager.GetWatchChannel
+
+// BindingReplicator replicates sfserviceinstance
```
Comment (anoopjb) [inline interoperator/controllers/multiclusterdeploy/sfservicebindingmetrics/sfservicebindingmetrics_controller.go:64]: I see the comments are outdated. Please update the inline comments.

File: interoperator/controllers/multiclusterdeploy/sfserviceinstancemetrics/sfserviceinstancemetrics_controller.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+// InstanceReplicator replicates a SFServiceInstance object to sister cluster
+type InstanceMetrics struct {
+	client.Client
+	Log             logr.Logger
+	clusterRegistry registry.ClusterRegistry
+	cfgManager      config.Config
+}
+
+// Reconcile reads that state of the cluster for a SFServiceInstance object on master and sister cluster
+// and replicates it.
```
[the remark this answers is not in the document]
Comment (anoopjb) [reply interoperator/controllers/multiclusterdeploy/sfserviceinstancemetrics/sfserviceinstancemetrics_controller.go:75]: I think these outdated comments will mislead the functionality of the method.

File: interoperator/controllers/multiclusterdeploy/sfserviceinstancemetrics/sfserviceinstancemetrics_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	mock_clusterRegistry "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/cluster/registry/mock_registry"
+	//"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+
+	"github.com/golang/mock/gomock"
+	"github.com/onsi/gomega"
+	//corev1 "k8s.io/api/core/v1"
+	//apierrors "k8s.io/apimachinery/pkg/api/errors"
+	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/apimachinery/pkg/types"
+	//"k8s.io/client-go/util/retry"
```
Comment (anoopjb) [inline interoperator/controllers/multiclusterdeploy/sfserviceinstancemetrics/sfserviceinstancemetrics_controller_test.go:35]:
```suggestion
```

I think it is good to remove the unused comments.

**Review by anoopjb — APPROVED** [review]
LGTM :+1:

**Pull request merged**